docs/proposals/application/README.md

```diff
@@ -0,0 +1,571 @@
+
+	// Select a Service.
+	ServiceRef *ServiceSelector `json:"serviceRef,omitempty"`
+
+	// Select an Ingress.
+	IngressRef *IngressSelector `json:"ingressRef,omitempty"`
+}
+
+// InfoItemSourceType is a string
+type InfoItemSourceType string
+
+// Constants for info type
+const (
+	SecretKeyRefInfoItemSourceType    InfoItemSourceType = "SecretKeyRef"
+	ConfigMapKeyRefInfoItemSourceType InfoItemSourceType = "ConfigMapKeyRef"
+	ServiceRefInfoItemSourceType      InfoItemSourceType = "ServiceRef"
+	IngressRefInfoItemSourceType      InfoItemSourceType = "IngressRef"
+)
+
+// ConfigMapKeySelector selects a key from a ConfigMap.
+type ConfigMapKeySelector struct {
+	// The ConfigMap to select from.
+	corev1.ObjectReference `json:",inline"`
+	// The key to select.
+	Key string `json:"key,omitempty"`
+}
+
+// SecretKeySelector selects a key from a Secret.
+type SecretKeySelector struct {
+	// The Secret to select from.
+	corev1.ObjectReference `json:",inline"`
+	// The key to select.
+	Key string `json:"key,omitempty"`
+}
+
+// ServiceSelector selects a Service.
+type ServiceSelector struct {
+	// The Service to select from.
+	corev1.ObjectReference `json:",inline"`
+	// The optional port to select.
+	Port *int32 `json:"port,omitempty"`
+	// The optional HTTP path.
+	Path string `json:"path,omitempty"`
+	// Protocol for the service
+	Protocol string `json:"protocol,omitempty"`
+}
+
+// IngressSelector selects an Ingress.
+type IngressSelector struct {
+	// The Ingress to select from.
+	corev1.ObjectReference `json:",inline"`
+	// The optional host to select.
+	Host string `json:"host,omitempty"`
+	// The optional HTTP path.
+	Path string `json:"path,omitempty"`
+	// Protocol for the ingress
+	Protocol string `json:"protocol,omitempty"`
+}
+```
+
+#### Example
+```
+apiVersion: application.karmada.io/v1alpha1
+kind: Application
+metadata:
+  name: test-1
+spec:
+  components:
+    - name: ref-deploy
+      type: objectTemplate
+      objectTemplate:
+        objectReference:
+          apiVersion: apps/v1
+          kind: Deployment
+          name: test
+          namespace: default
+        weight: 1
+    - name: ref-configmap
+      type: objectTemplate
+      objectTemplate:
+        objectReference:
+          apiVersion: v1
+          kind: Configmap
+          name: test
+          namespace: default
+        weight: 2
+status:
+  componentsReady: 2
+  componentsTotal: 2
+  components:
+    - identifier:
+        kind: Configmap
+        name: test
+        namespace: default
+        ordinal: 0
+        resource: ""
+        version: v1
+      status:
+        applied: true
+        healthy: true
+    - identifier:
+        group: app
+        kind: Deployment
+        name: nginx
+        namespace: default
+        ordinal: 1
+        resource: ""
+        version: v1
+      status:
+        applied: true
+        healthy: true
+```
+
+### Test Plan
+
+- Unit Test covering.
```

Comment:
Not sure what "covering" means here.
In my opinion, "Unit tests" is enough. Pls check.